badge_server/main.py

```diff
@@ -76,518 +63,195 @@ def _get_pair_status_for_packages(pkg_sets):
-    package_key = '{}_{}'.format(
-        package_name, commit_number) if commit_number else package_name
-
-    if badge_name is None:
-        badge_name = 'dependency status'
-
-    def run_check():
-        res = {
-            'status': 'UP_TO_DATE',
-            'details': {},
-            'timestamp': '',
-        }
-        details = {}
-        outdated = badge_utils.highlighter.check_package(package_name)
-        deprecated_deps_list = badge_utils.finder.get_deprecated_dep(
-            package_name)[1]
-        deprecated_deps = ', '.join(deprecated_deps_list)
-
-        max_level = badge_utils.priority_level.UP_TO_DATE
-        for dep in outdated:
-            dep_detail = {}
-            level = dep.priority.level
-            if level.value > max_level.value:
-                max_level = level
-            dep_detail['installed_version'] = dep.installed_version
-            dep_detail['latest_version'] = dep.latest_version
-            dep_detail['priority'] = dep.priority.level.name
-            dep_detail['detail'] = dep.priority.details
-            details[dep.name] = dep_detail
-            res['status'] = max_level.name
-            res['details'] = details
-            res['deprecated_deps'] = deprecated_deps
-        res['timestamp'] = datetime.datetime.now().strftime(
-            badge_utils.TIMESTAMP_FORMAT)
-
-        # Write the result to Cloud Datastore
-        cache.set('{}_dependency_badge'.format(package_key), res)
-
-    if not utils._is_package_in_whitelist([package_name]):
-        dependency_res = badge_utils._build_default_result(
-            badge_type=badge_utils.BadgeType.DEP_BADGE,
-            status='UNKNOWN',
-            details={})
-    else:
-        dependency_res = cache.get('{}_dependency_badge'.format(package_key))
-
-    if dependency_res is None:
-        details = badge_utils.DEFAULT_DEPENDENCY_RESULT
-    else:
-        details = dependency_res
-
-    # Run the check if dependency_res is None or forced to populate the cache
-    # or the cache is outdated.
-    if dependency_res is None or force_run_check is not None:
-        threading.Thread(target=run_check).start()
-    elif dependency_res is not None:
-        timestamp = dependency_res.get('timestamp')
-        if not badge_utils._is_github_cache_valid(timestamp):
-            threading.Thread(target=run_check).start()
-
-    badge = badge_utils._get_badge(details, badge_name)
-    response = flask.make_response(badge)
-    response.content_type = badge_utils.SVG_CONTENT_TYPE
-    response.headers['Cache-Control'] = 'no-cache'
-    response.add_etag()
+    # TODO: implement checks for the following conditions
+    # (they are currently using dummy info)
+    is_pypi = compat_utils._is_package_in_whitelist([pkgname])
+    is_latest_commit = True
+    if 'github.com' in pkgname:
+        parens_text = 'master' if is_latest_commit else 'updating...'
+        bdgname = '{} ({})'.format(bdgname, parens_text)
```

Comment:
Might be nicer to shorten the github.com package name, that's too long to display for now. Probably change it to 'github head (master)'

Reply:
done